src/code_memory/server.py

```
@@ -0,0 +1,199 @@
+        
+        ctx.info(f"Found {len(results)} matching snippets")
+        return results
+        
+    except Exception as e:
+        ctx.error(f"Search failed: {str(e)}")
+        logger.error("Failed to search code", exc_info=True)
+        return []
+
+
+@mcp.tool()
+async def list_languages(ctx: Context) -> List[Dict]:
```

Ruff (0.11.9)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)